Add KS test functionality for detecting tipping points

niatimary opened the pull request:
-  [x] Implements KS test for detecting tipping points using KS test
-  [x] Excludes climate_phase method (will be added in future PR)
-  [x] Includes test file for testing method functionality

niatimary committed:
Add KS test functionality for detecting tipping points

CommonClimate reviewed (changes requested):
Great! I did not run the tests but I trust that they run without issue. 

Small suggestions:

- the definition of `DeterministicTransitions` lists this reference:     Bagniewski, W., et al. (2021).  This is overly narrow as this class is meant to be extremely general. I don't believe we need a reference for it, either. 
- the code still imports the `ClimatePhases` class. Does it need to?
- the docstrings have inconsistent formatting. E.g. transitions.py uses ">>>" decorators, but series.py uses `.. jupyter-execute::`. Please use the latter throughout, as it will make it easier to build the documentation. 
- test_ks_return_types_t0() looks like it has an undefined variable called `ts`. Are you sure this runs without problem?

niatimary committed:
Fix code review suggestion

niatimary commented:
## Fixed Issues

1. DeterministicTransitions reference

2. Docstring formatting consistency
  - Updated all docstrings to use `.. jupyter-execute::` format consistently
  - Removed any `>>>` decorators to match the documentation build requirements

3. Test variable issue
  - Fixed the undefined `ts` variable in `test_ks_return_types_t0()`
  - Test now runs without problems (mistakenly removed the line earlier, now restored)

## Unchanged Issue
1. Keeping it for now; once I make the necessary changes and finalize a stable version, I’ll open another pull request.

CommonClimate reviewed (changes requested):
I don't understand why you are removing references 2 and 3 in `Series.kstest()`. Other than that, things look good.

niatimary commented:
I thought the first reference was more relevant than the other two, so I kept only that one. Do you think I should add the other references back as well?

CommonClimate committed:
put references back in
